netlogo-core/src/main/prim/_anyturtleson.scala

```diff
@@ -0,0 +1,45 @@
+// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo
+
+package org.nlogo.prim
+
+import org.nlogo.agent.{ AgentSet, Patch, Turtle }
+import org.nlogo.core.{ AgentKind, I18N, Syntax }
+import org.nlogo.nvm.{ ArgumentTypeException, Context, Reporter, RuntimePrimitiveException }
+
+class _anyturtleson extends Reporter {
+  override def report(context: Context): java.lang.Boolean =
+    Boolean.box(report_1(context, argEvalAgentSet(context, 0)))
```

Comment:
So this is where the failure is happening.  The `argEvalAgentSet()` doesn't allow the single patch (or turtle) through.  In the bytecode version (`report_1`), the argument `agentOrSet` is just `AnyRef` so it runs fine.  What we want here, then, is `args(0).report(context)` to get the `AnyRef` value for the `report_1` method, and it can throw the error if it's not a valid type for the `match`.  I should've noticed this on an earlier review, but hey, this is why the language tests are useful!